src/routes/safe/components/Transactions/TxList/SpendingLimitDetails.tsx

```diff
@@ -43,12 +43,16 @@ export const ModifySpendingLimitDetails = ({ txData, txInfo }: SpendingLimitProp
     [dataDecoded?.parameters],
   )
 
+  const beneficiaryValue = Array.isArray(beneficiary) ? (beneficiary[0] as string) : beneficiary
```

Comment:
When we determine the `beneficiary, tokenAddress, amount, resetTimeMin`we use `[]`as fallback.
Here we assume that the array always has length 1 and contains a string. Should we probably also check that the array has length 1 and throw / handle the unexpected data otherwise?

Reply:
Good point, currently all these values would be `undefined` in case of the fallback but for `amount` a fallback of 0 would make more sense and for `tokenAddress` an empty string.

Reply:
going one step deeper here, defaulting to an allowance with `amount` 0 is incorrect. I think we should omit the section where we render the "Amount"

Reply:
Makes total sense! I added the proper handling for the undefined fallbacks.

Reply:
resolving because this change was undone